### PR TITLE
Fixed AQGD optimizer groups objective function calls by default

### PR DESCRIPTION
**Pull request by proeseler**
### Summary
Fixes [#50](https://github.com/qiskit-community/qiskit-algorithms/issues/50#issue-1861052272)

It has been fixed that the AQGD optimizer groups objective function calls by default.

### Details and comments
The number of requests is chunked in a similar way to the gradient_num_diff() method from the Optimizer class. The number of grouped evaluations can be controlled via the max_grouped_evals parameter now and a test has been added to check whether the new functionality works for a parallelizable and a non-parallelizable objective function.

### CODE REVIEW

File: qiskit_algorithms/optimizers/aqgd.py
Comment (woodsp-ibm) [inline qiskit_algorithms/optimizers/aqgd.py:60]: This should be added to the Args: docstring below. And I think we should note that max_evals_grouped will max out at 3 such that even if a larger number is passed the max evals that can be grouped is 3.

File: releasenotes/notes/fix_aqgd_max_grouped_evals-fbe108c005a9b7ac.yaml
Comment (woodsp-ibm) [inline releasenotes/notes/fix_aqgd_max_grouped_evals-fbe108c005a9b7ac.yaml:5]: Maybe we can elaborate on this so that someone reading the release note, not knowing the issue, things are clear. Something like this perhaps - feel free to re-word etc.
```
It has been fixed that the AQGD optimizer groups objective function calls by default so that now
it passed a single point to the objective function. For algorithms that can handle a more than
one point in their objective function, such as a VQE in the algorithms here, the number of grouped
evaluations can be controlled via the max_grouped_evals parameter. Grouping evals allows a list
of points to be passed so that potentially they may be more efficiently evaluated in a single job.
```

File: test/optimizers/test_optimizer_aqgd.py
Comment (woodsp-ibm) [inline test/optimizers/test_optimizer_aqgd.py:112]: Can you make this bit where you use the same VQE, but with different evals grouped a subtest. Or perhaps better split this VQE test out from the test below. They are more independent tests and then this test could be parameterized with using `ddt` eg heres an example, https://github.com/qiskit-community/qiskit-algorithms/blob/9f85dbcee08a80ff82ba82bac2ae19e06b3237ee/test/optimizers/test_optimizers.py#L187

This way you could test for 1, 2 and 3 and `ddt' via, the decorator, will make 3 separate tests - you would see that when running as it adds a suffice to the test name for each it creates.

Having separate tests is better, whether by separate functions or subTest. When they are separate then if the test fails, or say in an earlier assert in the same function with subTest later testing is still run. So as it stands here maybe it fails with grouped of 2, would it pass with VQE one, and what about the custom object test where it checks the the point, would that have provided some info had it run. Having other parts, that are logically independent, run and knowing if they passed or failed may give extra information that may help identify the failure.

**Comment by woodsp-ibm** [comment]
Note: CI is failing here for Python 3.12 - see #113 

As to `Algorithms Unit Tests / Monitor_Qiskit_Main (ubuntu-latest, 3.8) ` this is one a pair that test against the main branch in Qiskit, rather than stable release, so we can monitor for breaking changes as Qiskit develops. Neither of this monitor pair are required to pass for the PR (you can see they do not have the Required label. But it is something that will need investigating as to why this is happening as there should not be breaking changes like this - it just recently started it seems, the #113 has been going on a few days.

File: releasenotes/notes/fix_aqgd_max_grouped_evals-fbe108c005a9b7ac.yaml
Comment (proeseler) [reply releasenotes/notes/fix_aqgd_max_grouped_evals-fbe108c005a9b7ac.yaml:5]: Makes sense. I will do so.

File: qiskit_algorithms/optimizers/aqgd.py
Comment (proeseler) [reply qiskit_algorithms/optimizers/aqgd.py:60]: I have just two short question regarding the second part. Maybe I misunderstood the question, but shouldn't it max out at 2*n+1, if n is the dimension of x0? And shouldn't it then also be specified for the method `gradient_num_diff` in the class [Optimizer(ABC)](https://docs.quantum.ibm.com/api/qiskit/qiskit.algorithms.optimizers.Optimizer)?

Comment (proeseler) [reply qiskit_algorithms/optimizers/aqgd.py:60]: Regarding the first part. Makes sense. I will do so 👍

File: test/optimizers/test_optimizer_aqgd.py
Comment (proeseler) [reply test/optimizers/test_optimizer_aqgd.py:112]: Makes sense. I will do so.

File: qiskit_algorithms/optimizers/aqgd.py
Comment (woodsp-ibm) [reply qiskit_algorithms/optimizers/aqgd.py:60]: >  but shouldn't it max out at 2*n+1

Yes, you are right. I was confusing things. SPSA does have limit since it perturbs all parameters together for + and - but for no limit in this case its fine as is - whatever the limit its down to the optimizer subclass - normally it scales some way based on num parameters (dimension of point) like this.

Comment (proeseler) [reply qiskit_algorithms/optimizers/aqgd.py:60]: I added the changes. Please take a look if that fits.

**Review by ElePT — APPROVED** [review]
The changes look good to me, thansk @proeseler for the contribution. I'll approve but wait for @woodsp-ibm's confirmation before merging.

**Review by woodsp-ibm — APPROVED** [review]
Thanks for doing this and taking care of the comments. It all looks good to me - sorry for the delay in getting back and reviewing this.

**Pull request merged**
